### PR TITLE
Docs/more fixes

### LINKED ISSUE #2901: Link to dask documentation on chunks
It would be good to link to https://docs.dask.org/en/latest/array-chunks.html in https://xarray.pydata.org/en/stable/dask.html#chunking-and-performance

### LINKED ISSUE #2908: More efficient rolling with large dask arrays
#### Code Sample

```python
import xarray as xr
import dask.array as da

dsize=[62,12,100,192,288]
array1=da.random.random(dsize,chunks=(dsize[0],dsize[1],1,dsize[3],int(dsize[4]/2)))
array2=xr.DataArray(array1)
rollingmean=array2.rolling(dim_1=3,center=True).mean()  # <-- this kills all workers

```
#### Problem description

I'm working on NCAR's cheyenne with a 36GB netcdf using dask_jobqueue.PBSCluster, and trying to calculate the running-mean along one dimension. Despite having plenty of memory reserved (400GB), I can watch DataArray.rolling blow up the bytes stored in the dashboard until the job hangs and all the workers are killed.  

The above snippet reproduces the issue with the same array size and chunksize as what I'm working with. This worker-killing behavior does not occur for arrays that are 100x smaller.  I've found a speedy way to calculate what I need without using rolling, but I thought I should bring this to your attention regardless. 

In case it's relevant, here's how I'm setting up the dask cluster on cheyenne:
```python
from dask.distributed import Client
from dask_jobqueue import PBSCluster  #version 0.4.1

cluster=PBSCluster(cores=36, processes=9, memory='109GB', project=myproj, resource_spec='select=1:ncpus=36:mem=109G', queue='regular', walltime='02:00:00')
numnodes=4
client = Client(cluster)
cluster.scale(numnodes*9)

```

#### Output of ``xr.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: None
python: 3.7.1 (default, Dec 14 2018, 19:28:38) 
[GCC 7.3.0]
python-bits: 64
OS: Linux
OS-release: 3.12.62-60.64.8-default
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8
libhdf5: 1.10.4
libnetcdf: 4.6.2

xarray: 0.12.1
pandas: 0.24.1
numpy: 1.15.4
scipy: 1.2.1
netCDF4: 1.4.2
pydap: None
h5netcdf: None
h5py: None
Nio: None
zarr: 2.3.1
cftime: 1.0.3.4
nc_time_axis: None
PseudonetCDF: None
rasterio: None
cfgrib: None
iris: None
bottleneck: None
dask: 1.1.5
distributed: 1.26.1
matplotlib: 3.0.2
cartopy: 0.17.0
seaborn: 0.9.0
setuptools: 40.6.3
pip: 18.1
conda: 4.6.13
pytest: None
IPython: 7.3.0
sphinx: None

</details>

### LINKED ISSUE #2909: docs questions from new user
A relatively new user here gave me an incomplete list of things they found to be confusing in the docs:

-  ~What is xarray's data model?~ opened #2913 
- [x] What is split-apply-combine?
- [x] What are dummy dimensions and `np.newaxis`

- [x] quick-overiew: Index so that you extract a subset!

- [ ] too many references to pandas!
- [x] do I need to know pandas to use xarray?
- [ ] what is the difference between attrs and encoding? How do I use them properly/correctly?
- [ ] why `isnull` and `dropna`: why `dropna` instead of `dropnull`
- [x] `compute` docstring doesn't mention computation!
- [x] compute vs persist vs load: what's the difference
- [ ] do I get parallelization for free?
  - [ ] do I start a dask cluster first?
  - [x] dask page links to shoyers blog but that doesn't explicitly start a cluster.

I'm crossing out things that I've started to fix.

### PR DESCRIPTION
**Pull request by dcherian**
- partially addresses #2909 , closes #2901, closes #2908

### CODE REVIEW

File: doc/dask.rst [hunk 1]
```diff
@@ -355,30 +397,6 @@ application.
     structure of a problem, unlike the generic speedups offered by
     ``dask='parallelized'``.
 
-Chunking and performance
```
Comment (dcherian) [inline doc/dask.rst]: combined with earlier chunk stuff

File: doc/extending.rst [hunk 2]
```diff
@@ -0,0 +1,90 @@
+.. _extending:
+
```
Comment (dcherian) [inline doc/extending.rst]: Moved "extending xarray" one level up

File: doc/io.rst [hunk 3]
```diff
@@ -15,82 +15,6 @@ format (recommended).
     import xarray as xr
     np.random.seed(123456)
 
-.. _io.pickle:
```
Comment (dcherian) [inline doc/io.rst]: moved netCDF up for emphasis.

File: doc/quick-overview.rst [hunk 4]
```diff
@@ -48,21 +48,21 @@ Here are the key properties for a ``DataArray``:
 Indexing
 --------
 
-xarray supports four kind of indexing. Since we have assigned coordinate labels to the x dimension we can use label-based indexing along that dimension just like pandas. The four examples below all yield the same result but at varying levels of convenience and intuitiveness.
+xarray supports four kind of indexing. Since we have assigned coordinate labels to the x dimension we can use label-based indexing along that dimension just like pandas. The four examples below all yield the same result (the value at `x=10`) but at varying levels of convenience and intuitiveness.
 
 .. ipython:: python
 
     # positional and by integer label, like numpy
```
Comment (dcherian) [inline doc/quick-overview.rst]: changed so that we are actually subsetting

File: doc/why-xarray.rst [hunk 5]
```diff
@@ -18,13 +22,14 @@ Xarray doesn't just keep track of labels on arrays -- it uses them to provide a
 powerful and concise interface. For example:
 
 -  Apply operations over dimensions by name: ``x.sum('time')``.
--  Select values by label instead of integer location:
+-  Select values by label (or logical location) instead of integer location:
    ``x.loc['2014-01-01']`` or ``x.sel(time='2014-01-01')``.
 -  Mathematical operations (e.g., ``x - y``) vectorize across multiple
    dimensions (array broadcasting) based on dimension names, not shape.
--  Flexible split-apply-combine operations with groupby:
+-  Easily use the `split-apply-combine <https://vita.had.co.nz/papers/plyr.pdf>`_
```
Comment (dcherian) [inline doc/why-xarray.rst]: Is there a better link for split-apply-combine?

**Comment by dcherian** [comment]
Edited to reduce the number of changes so that it's easier to review.

File: doc/dask.rst [hunk 6]
```diff
@@ -5,13 +5,13 @@ Parallel computing with Dask
...
 computations and streaming computation on datasets that don't fit into memory.
-
 Currently, Dask is an entirely optional feature for xarray. However, the
 benefits of using Dask are sufficiently strong that Dask may become a required
 dependency in a future version of xarray.
 
 For a full example of how to use xarray's Dask integration, read the
-`blog post introducing xarray and Dask`_.
+`blog post introducing xarray and Dask`_. More up-to-date examples
+may be found at the `Pangeo project's use-cases <http://pangeo.io/use_cases/index.html>`_.
```
[the remark this answers is not in the document]
Comment (jhamman) [reply doc/dask.rst]: We can link to the dask-examples repo: https://examples.dask.org/xarray.html

**Comment by jhamman** [comment]
Thanks @dcherian!

**Comment by max-sixty** [comment]
Thanks @dcherian !

**Pull request merged**